### PR TITLE
chore(Python): add depth first search

### PR DESCRIPTION
**Pull request by PrathameshSahasrabuddhe**
**PR Checklist:**
- [x] My submission is formatted according to the guidelines in the [contributing guide](https://github.com/MakeContributions/DSA#contribution-guidelines)
- [x] My addition is on refer on the language README.md file
- [x] My addition does not have a spelling problem
- [x] My submission has a proper and user-friendly description of the algorithm
- [x] My submission has the time complexity of the algorithm
- [x] My submission has sample input-output of the program (NOT FOR PYTHON)

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**

I've added the Depth First Search algorithm in Python language. I have used the recursive implementation.
A link is also available for the algorithm in the README.md file in the Python folder under Recursion heading

**Other information:**

### CODE REVIEW

File: algorithms/Python/recursion/depth-first-search.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# A dfs alorithm in Python
```
Comment (Arsenic-ATG) [inline algorithms/Python/recursion/depth-first-search.py:1]: typo detected by codespell

alorithm ==> algorithm
```suggestion
# A dfs algorithm in Python
```

File: algorithms/Python/README.md [hunk 2]
```diff
@@ -26,6 +26,7 @@
 - [n-th Fibonacci number](recursion/nth_fibonacci_number.py)
 - [Recursive Insertion Sort](recursion/recursive_insertion_sort.py)
 - [Recursive Sum of n numbers](recursion/recursive-sum-of-n-numbers.py)
+- [Depth First Search](recursion/depth-first-search.py)
```
Comment (Arsenic-ATG) [inline algorithms/Python/README.md:29]: I think a better place for dfs should be under graph section rather than under recursion section

**Review by Arsenic-ATG — CHANGES_REQUESTED** [review]

File: algorithms/Python/README.md (hunk 2 above)
Comment (PrathameshSahasrabuddhe) [reply algorithms/Python/README.md:29]: Sorry I didn't think about it before. I will change change that. Thank you for pointing that out.

**Comment by PrathameshSahasrabuddhe** [comment]
When will you merge this code?

**Pull request merged**